### PR TITLE
Various Variant reporting IGV improvements

### PR DESCRIPTION
**Pull request by susannasiebert**
This PR addresses the following issues:
https://jira.gsc.wustl.edu/browse/AT-717
https://jira.gsc.wustl.edu/browse/AT-718
https://jira.gsc.wustl.edu/browse/AT-719

### CODE REVIEW

File: lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        },
+        output_file => {
+            type => 'String',
+            is_optional => 0,
+            doc => 'Output XML file',
+        },
+        reference_name => {
+            type => 'String',
+            is_optional => 0,
+            doc => 'the name of the reference (in IGV) that the bams are aligned to. E.g. "b37" for build 37 or "reference"for our internal build36',
```
Comment (tmooney) [inline lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm]: There's a space missing in `"reference"for`.  (Also `build 37` vs. `build36` is inconsistent.)

These could also be listed as `example_values => ['b37', 'reference']`.

File: lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    ]
+};
+
+sub help_brief {
+    "Makes an IGV session for review. Supports as many tracks as you want. Panels and track layout are not explicitly specified."
+}
+
+sub help_detail {
+    <<'HELP';
+This helps create IGV session files for manual reviewers
```
Comment (tmooney) [inline lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm]: Should the brief and the detail be switched? It's unusual for the "brief" version to be longer than the "detail" version.

File: lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        OUTPUT => $output_fh,
+        NEWLINES => 1,
+        ENCODING => 'utf-8'
+    );
+
+    $xml_writer->xmlDecl('UTF-8', 'no');
+    $xml_writer->startTag(
+        'Session',
+        genome => $self->reference_name,
+        locus => '1:1-100',
```
Comment (tmooney) [inline lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm]: Does this tool have limitations on what references it accepts?  (What happens if the chosen reference doesn't have a chromosome `1` but instead, say, `chr1`?)

File: lib/perl/Genome/VariantReporting/Framework/Component/Report/MergedReport.pm [hunk 4]
```diff
@@ -540,5 +541,25 @@ sub get_entry_source {
...
 
+sub category {
+    my $self = shift;
+
+    my @report_users = map { $_->users('label like' => 'report:%') } $self->report_results;
+    my $category;
+    for my $user (@report_users) {
+        if ($user->label =~ /report:(.*)/) {
+            my $metadata_json = $1;
+            my $m = from_json($metadata_json);
```
Comment (tmooney) [inline lib/perl/Genome/VariantReporting/Framework/Component/Report/MergedReport.pm]: I'm late to the game, but it seems like JSON would be better stored elsewhere (a SoftwareResult::Metric perhaps?) than in a user label.

Comment (davidlmorton) [inline lib/perl/Genome/VariantReporting/Framework/Component/Report/MergedReport.pm]: That is something we should discuss in more detail, I agree (on the face of it) that it seems better to have that information live in a Metric instead of a User label.  

Perhaps the `DelegatesToResult` class could take an argument that would create the metrics in almost the same manner as it takes a `user` and a `label` argument now.

We'll have to think about migration too if we decide to switch it.

File: lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm (hunk 3 above)
Comment (susannasiebert) [inline lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm]: That is a very good question and I don't know the answer off-hand. For what it's worth, this tools is based on the other DumpIgvXml tools which also use 1:1-100.

**Comment by susannasiebert** [comment]
@tmooney I pushed a commit to address your comments about the documentation.

File: lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm (hunk 3 above)
Comment (susannasiebert) [inline lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm]: I did some trial runs on an example igv session xml using variations of b37/h18 and 1:1-100/ch1:1-100 and it seems to translate between the two without problems.

File: lib/perl/Genome/VariantReporting/Framework/Component/Report/MergedReport.pm (hunk 4 above)
Comment (susannasiebert) [inline lib/perl/Genome/VariantReporting/Framework/Component/Report/MergedReport.pm]: I think we can agree that this is outside of the scope of this PR.

**Comment by susannasiebert** [comment]
I pushed another commit to clean up the igv reference names since there were some inconsistencies/errors.

File: lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+use Genome;
+use Command;
+use IO::File;
+use Cwd qw(abs_path);
+use URI;
+use XML::Writer;
+
+class Genome::Model::Tools::Analysis::DumpIgvXmlBasic {
+    is => 'Command',
```
Comment (davidlmorton) [inline lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm]: Should this be Command::V2?

File: lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    has => [
+        resource_files => {
+            type => 'HASH',
```
Comment (davidlmorton) [inline lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm]: Does Command / Command::V2 support HASH type arguments to be supplied via the command line?

**Comment by davidlmorton** [comment]
+1 with Command -> Command:V2 change or some reason why it isn't changed. :)

File: lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm (hunk 6 above)
Comment (susannasiebert) [inline lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm]: I honestly don't know. I think these igv tools are primarily called inside of Perl code, not on the command line. @tmooney do you know?

What's the alternative here if we can't use HASH?

File: lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm (hunk 5 above)
Comment (susannasiebert) [inline lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm]: My bad. This is what happens when you just copy things from existing tools.

File: lib/perl/Genome/VariantReporting/Framework/Component/Report/MergedReport.pm (hunk 4 above)
Comment (tmooney) [inline lib/perl/Genome/VariantReporting/Framework/Component/Report/MergedReport.pm]: It definitely is.  Should we make a JIRA issue about it?

File: lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm (hunk 5 above)
Comment (susannasiebert) [inline lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm]: This is now a Command:V2

File: lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm (hunk 6 above)
Comment (susannasiebert) [inline lib/perl/Genome/Model/Tools/Analysis/DumpIgvXmlBasic.pm]: This has been addressed by setting _is_hidden_in_docs to 1 which results in this command not showing up in the docs/autocomplete of the command line. People can still try to execute it on the command line but they would need to know that it exists.

File: lib/perl/Genome/VariantReporting/Framework/Component/Report/MergedReport.pm (hunk 4 above)
Comment (susannasiebert) [inline lib/perl/Genome/VariantReporting/Framework/Component/Report/MergedReport.pm]: Yes. I made one here https://jira.gsc.wustl.edu/browse/AT-731

**Comment by tmooney** [comment]
+1 assuming tests pass! :dolphin:
